pkg/resource/statefulset.go

```diff
@@ -247,6 +247,13 @@ func (b StatefulSetBuilder) MakeContainers() []corev1.Container {
 			Resources:       b.Spec().Resources,
 			Command:         b.commandArgs(),
 			Env:             b.envVars(),
+			Lifecycle: &corev1.Lifecycle{
+				PreStop: &corev1.Handler{
+					Exec: &corev1.ExecAction{
+						Command: []string{"/cockroach/cockroach", "node", "drain", b.SecureMode()},
```

Comment:
@chrislovecnm have you gotten this to executre successfully? @alinadonisa and I couldn't get it to run for some reason 🤔 

I would also recommend wrapping this in `sh -c "%s || exit 0"`. If for whatever reason a lifecycle hook exits with a non-zero status code, the pod will be `SIGKILL`'d which is the last thing that we want if draining fails...

> If either a PostStart or PreStop hook fails, it kills the Container.

https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/#hook-handler-execution

Reply:
what is the full string that I should be using?

Reply:
Got it from @chrisseto on slack